doc/source/operations/gpu-in-openstack.rst

```diff
@@ -0,0 +1,1122 @@
+    /opt/dell/srvadmin/bin/idracadm7 jobqueue create BIOS.Setup.1-1
+    <reboot>
+
+
+Obtain driver from NVIDIA licensing portal
+-------------------------------------------
+
+Download Nvidia GRID driver from `here <https://docs.nvidia.com/grid/latest/grid-software-quick-start-guide/index.html#redeeming-pak-and-downloading-grid-software>`__
+(This requires a login). The file can either be placed on the :ref:`ansible control host<NVIDIA control host>` or :ref:`uploaded to pulp<NVIDIA Pulp>`.
+
+.. _NVIDIA Pulp:
+
+Uploading the GRID driver to pulp
+---------------------------------
+
+Uploading the driver to pulp will make it possible to run kayobe from any host. This can be useful when
+running in a CI environment.
+
+.. code:: shell
+
+    pulp artifact upload --file ~/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip
+    pulp file content create --relative-path "NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip" --sha256 c8e12c15b881df35e618bdee1f141cbfcc7e112358f0139ceaa95b48e20761e0
+    pulp file repository create --name nvidia
+    pulp file repository content add --repository nvidia --sha256 c8e12c15b881df35e618bdee1f141cbfcc7e112358f0139ceaa95b48e20761e0 --relative-path "NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+    pulp file publication create --repository nvidia
+    pulp file distribution create --name nvidia --base-path nvidia --repository nvidia
+
+The file will then be available at ``<pulp_url>/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip``. You
+will need to set the ``vgpu_driver_url`` configuration option to this value:
+
+.. code:: yaml
+
+   # URL of GRID driver in pulp
+   vgpu_driver_url: "{{ pulp_url }}/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA control host:
+
+Placing the GRID driver on the ansible control host
+---------------------------------------------------
+
+Copy the driver bundle to a known location on the ansible control host. Set the ``vgpu_driver_url`` configuration variable to reference this
+path using ``file`` as the url scheme e.g:
+
+.. code:: yaml
+
+    # Location of NVIDIA GRID driver on localhost
+    vgpu_driver_url: "file://{{ lookup('env', 'HOME') }}/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA OS Configuration:
+
+OS Configuration
+----------------
+
+Host OS configuration is done by using roles in the `stackhpc.linux <https://github.com/stackhpc/ansible-collection-linux>`_ ansible collection.
+
+Add the following to your ansible ``requirements.yml``:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/ansible/requirements.yml
+
+    #FIXME: Update to known release When VGPU and IOMMU roles have landed
+    collections:
+      - name: stackhpc.linux
+        source: git+https://github.com/stackhpc/ansible-collection-linux.git,preemptive/vgpu-iommu
```

Comment:
Do we still need to specify branch with stackhpc.linux collection?

Reply:
@jovial do you know what the state of vgpu and iommu support is? I see there's a couple of open PRs for this, but none for preemptive/vgpu-iommu

Reply:
It's already in Kayobe upstream (in antelope) so not required here

Reply:
See https://review.opendev.org/c/openstack/kayobe/+/887200
There's a lot we can remove from this page now, we should instead link to the [upstream docs](https://docs.openstack.org/kayobe/latest/configuration/reference/vgpu.html)